### ISSUE TITLE
Unable to use Base.OneTo as indices in LinearInterpolation

### ISSUE THREAD

**Issue by jishnub** [JuliaMath/Interpolations.jl#325]
I encounter the following error 
```
julia> a=collect(1:5);

julia> Base.OneTo(length(a)) == 1:length(a)
true

julia> LinearInterpolation(1:length(a),a)
5-element extrapolate(scale(interpolate(::Array{Float64,1}, BSpline(Linear())), (1:5,)), Throw()) with element type Float64:
 1.0
 2.0
 3.0
 4.0
 5.0

julia> LinearInterpolation(Base.OneTo(length(a)),a)
5-element extrapolate(scale(interpolate(::Array{Float64,1}, BSpline(Linear())), (Base.OneTo(5),)), Throw()) with element type Float64:
Error showing value of type Interpolations.Extrapolation{Float64,1,ScaledInterpolation{Float64,1,Interpolations.BSplineInterpolation{Float64,1,Array{Float64,1},BSpline{Linear},Tuple{Base.OneTo{Int64}}},BSpline{Linear},Tuple{Base.OneTo{Int64}}},BSpline{Linear},Throw{Nothing}}:
ERROR: MethodError: no method matching coordlookup(::Base.OneTo{Int64}, ::Int64)
Closest candidates are:
  coordlookup(::NoInterp, ::Any, ::Any) at /home/jb6888/.julia/packages/Interpolations/6RNnw/src/scaling/scaling.jl:88
  coordlookup(::Interpolations.Flag, ::Any, ::Any) at /home/jb6888/.julia/packages/Interpolations/6RNnw/src/scaling/scaling.jl:89
  coordlookup(::UnitRange, ::Any) at /home/jb6888/.julia/packages/Interpolations/6RNnw/src/scaling/scaling.jl:91
  ...
Stacktrace:
 [1] coordlookup(::BSpline{Linear}, ::Base.OneTo{Int64}, ::Int64) at /home/jb6888/.julia/packages/Interpolations/6RNnw/src/scaling/scaling.jl:89
 [2] coordslookup at /home/jb6888/.julia/packages/Interpolations/6RNnw/src/scaling/scaling.jl:83 [inlined]
 [3] gradient at /home/jb6888/.julia/packages/Interpolations/6RNnw/src/scaling/scaling.jl:105 [inlined]
 [4] Extrapolation at /home/jb6888/.julia/packages/Interpolations/6RNnw/src/extrapolation/extrapolation.jl:44 [inlined]
 [5] (::getfield(Interpolations, Symbol("##46#47")){Interpolations.Extrapolation{Float64,1,ScaledInterpolation{Float64,1,Interpolations.BSplineInterpolation{Float64,1,Array{Float64,1},BSpline{Linear},Tuple{Base.OneTo{Int64}}},BSpline{Linear},Tuple{Base.OneTo{Int64}}},BSpline{Linear},Throw{Nothing}}})(::Tuple{Int64}) at ./none:0
 [6] iterate at ./generator.jl:47 [inlined]
 [7] collect at ./array.jl:606 [inlined]
 [8] show_ranged(::IOContext{REPL.Terminals.TTYTerminal}, ::Interpolations.Extrapolation{Float64,1,ScaledInterpolation{Float64,1,Interpolations.BSplineInterpolation{Float64,1,Array{Float64,1},BSpline{Linear},Tuple{Base.OneTo{Int64}}},BSpline{Linear},Tuple{Base.OneTo{Int64}}},BSpline{Linear},Throw{Nothing}}, ::Tuple{Base.OneTo{Int64}}) at /home/jb6888/.julia/packages/Interpolations/6RNnw/src/io.jl:107
 [9] show(::IOContext{REPL.Terminals.TTYTerminal}, ::MIME{Symbol("text/plain")}, ::Interpolations.Extrapolation{Float64,1,ScaledInterpolation{Float64,1,Interpolations.BSplineInterpolation{Float64,1,Array{Float64,1},BSpline{Linear},Tuple{Base.OneTo{Int64}}},BSpline{Linear},Tuple{Base.OneTo{Int64}}},BSpline{Linear},Throw{Nothing}}) at /home/jb6888/.julia/packages/Interpolations/6RNnw/src/io.jl:117
 [10] display(::REPL.REPLDisplay, ::MIME{Symbol("text/plain")}, ::Any) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:131
 [11] display(::REPL.REPLDisplay, ::Any) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:135
 [12] display(::Any) at ./multimedia.jl:287
 [13] #invokelatest#1 at ./essentials.jl:742 [inlined]
 [14] invokelatest at ./essentials.jl:741 [inlined]
 [15] print_response(::IO, ::Any, ::Any, ::Bool, ::Bool, ::Any) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:155
 [16] print_response(::REPL.AbstractREPL, ::Any, ::Any, ::Bool, ::Bool) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:140
 [17] (::getfield(REPL, Symbol("#do_respond#38")){Bool,getfield(REPL, Symbol("##48#57")){REPL.LineEditREPL},REPL.LineEditREPL,REPL.LineEdit.Prompt})(::Any, ::Any, ::Any) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:714
 [18] #invokelatest#1 at ./essentials.jl:742 [inlined]
 [19] invokelatest at ./essentials.jl:741 [inlined]
 [20] run_interface(::REPL.Terminals.TextTerminal, ::REPL.LineEdit.ModalInterface, ::REPL.LineEdit.MIState) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.1/REPL/src/LineEdit.jl:2273
 [21] run_frontend(::REPL.LineEditREPL, ::REPL.REPLBackendRef) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:1035
 [22] run_repl(::REPL.AbstractREPL, ::Any) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:192
 [23] (::getfield(Base, Symbol("##734#736")){Bool,Bool,Bool,Bool})(::Module) at ./client.jl:362
 [24] #invokelatest#1 at ./essentials.jl:742 [inlined]
 [25] invokelatest at ./essentials.jl:741 [inlined]
 [26] run_main_repl(::Bool, ::Bool, ::Bool, ::Bool, ::Bool) at ./client.jl:346
 [27] exec_options(::Base.JLOptions) at ./client.jl:284
 [28] _start() at ./client.jl:436
```

This would be convenient as we can pass axes of arrays as the indices

**Comment by tomasaschan** [comment]
Here are the existing methods of `coordlookup`:
https://github.com/JuliaMath/Interpolations.jl/blob/d1ad2a1409ce6cea96cd304a540c485d2b393f5d/src/scaling/scaling.jl#L88-L101

IIUC, given that `Base.OneTo` [inherits `AbstractUnitRange`](https://github.com/JuliaLang/julia/blob/95a52a5d6210ce07b54f1c8cb74eea88f0cd9eee/base/range.jl#L307), we should be able to make this work just by changing the `UnitRange` method to take an `AbstractUnitRange` (and use accessor methods instead of field access to get the `start` and `end` values). Care to file a PR? :)

**Comment by jishnub** [comment]
This should be fixed by #327
